src/utils/src/ExponentialBackoffRetryStrategy.c

```diff
@@ -232,6 +232,20 @@ STATUS getExponentialBackoffRetryStrategyWaitTime(PRetryStrategy pRetryStrategy,
     return retStatus;
 }
 
+STATUS getExponentialBackoffRetryCount(PRetryStrategy pRetryStrategy, PUINT32 retryCount) {
+    ENTERS();
+    STATUS retStatus = STATUS_SUCCESS;
+    PExponentialBackoffRetryStrategyState pRetryState = NULL;
+    CHK_STATUS(pRetryStrategy == NULL, STATUS_NULL_ARG);
+    pRetryState = TO_EXPONENTIAL_BACKOFF_STATE(pRetryStrategy);
+    MUTEX_LOCK(pRetryState->retryStrategyLock);
```

Comment:
is it possible that `pRetryState` is `NULL`?

Reply:
It is, I am doing a NULL check to avoid any dereferencing.